pkg/kubelet/dockertools/docker_manager_linux.go

```diff
@@ -40,8 +40,13 @@ func getContainerIP(container *dockertypes.ContainerJSON) string {
 func getNetworkingMode() string { return "" }
 
 // Returns true if the container name matches the infrastructure's container name
-func containerProvidesPodIP(name *KubeletContainerName) bool {
-	return name.ContainerName == PodInfraContainerName
+func containerProvidesPodIP(containerName string) bool {
```

Comment:
What's the difference between `containerProvidesPodIP` and `containerIsNetworked`, couldn't these just be a single function?